bot/exts/filtering/_settings_types/actions/ping.py

```diff
@@ -0,0 +1,47 @@
+from typing import ClassVar
+
+from pydantic import validator
+
+from bot.exts.filtering._filter_context import FilterContext
+from bot.exts.filtering._settings_types.settings_entry import ActionEntry
+from bot.exts.filtering._utils import resolve_mention
+
+
+class Ping(ActionEntry):
+    """A setting entry which adds the appropriate pings to the alert."""
+
+    name: ClassVar[str] = "mentions"
+    description: ClassVar[dict[str, str]] = {
+        "guild_pings": (
+            "A list of role IDs/role names/user IDs/user names/here/everyone. "
+            "If a mod-alert is generated for a filter triggered in a public channel, these will be pinged."
+        ),
+        "dm_pings": (
+            "A list of role IDs/role names/user IDs/user names/here/everyone. "
+            "If a mod-alert is generated for a filter triggered in DMs, these will be pinged."
+        )
+    }
+
+    guild_pings: set[str]
+    dm_pings: set[str]
+
+    @validator("*", pre=True)
+    @classmethod
+    def init_sequence_if_none(cls, pings: list[str] | None) -> list[str]:
+        """Initialize an empty sequence if the value is None."""
+        if pings is None:
+            return []
+        return pings
+
+    async def action(self, ctx: FilterContext) -> None:
+        """Add the stored pings to the alert message content."""
+        mentions = self.guild_pings if not ctx.channel or ctx.channel.guild else self.dm_pings
+        new_content = " ".join([resolve_mention(mention) for mention in mentions])
+        ctx.alert_content = f"{new_content} {ctx.alert_content}"
+
+    def __or__(self, other: ActionEntry):
+        """Combines two actions of the same type. Each type of action is executed once per filter."""
+        if not isinstance(other, Ping):
```

Comment:
Why does the type hint for `other` not match the guard condition for `NotImplemented` here?

Reply:
This is something I wasn't very sure about. I didn't want to change the signature of the overridden method. Is changing the type hint considered common?

Reply:
@mbaruh If you want to require the same class, use `typing.Self` in the parent and child classes instead of `ActionPing`.

Also, it's still a good idea to check for `isinstance(other, Ping)` because the right operand might implement `__ror__`...

Reply:
Actually, why is this a dunder method and not just a normal method? Dunder methods have quite a lot of hariballish semantics to them (like `NotImplemented`).

Reply:
`typing.Self` is 3.11, the bot uses 3.10.

Being able to use `|` for this operation just seemed nice here 😅

Reply:
> `typing.Self` is 3.11, the bot uses 3.10.

Import it from `typing_extensions`

>  seemed nice

Between nicity and simplicity I would surely pick simplicity :)

Reply:
Fair point. When would you say it would be worth using the dunder then?

Reply:
Not sure actually. It is pretty nice if you're making some super general data structure, like a `set` or `enum.IntFlag`. IMO dunders are useful for writing more complex DSL-like expressions, like `foo | (bar & baz - (fizz | buzz))`.

Although Clojure programmers are fine without infix notation at all :shrug: 
```clj
(union foo
  (difference
    (intersection bar baz)
    (union fizz buzz)))
```